### PR TITLE
feat: [CI-9685]: Add support in CI Manager to fetch OIDC ID Token from NG Manager

### PR DESCRIPTION
**Pull request by Raghav-Murali**
## Describe your changes

This PR adds support in CI Manager to fetch the OIDC ID Token from NG Manager.

We have verified this working E2E in the local setup.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54587)

### CODE REVIEW

File: 332-ci-manager/config/ci-manager-config.yml [hunk 1]
```diff
@@ -366,8 +366,8 @@ ciExecutionServiceConfig:
       buildAndPushDockerRegistry: plugins/docker:20.14.5
       buildAndPushACR: plugins/acr:20.14.5
       buildAndPushECR: plugins/ecr:20.14.5
-      buildAndPushGCR: plugins/gcr:20.14.5
-      buildAndPushGAR: plugins/gcr:20.15.0
+      buildAndPushGCR: plugins/gcr:latest
```
Comment (eoinmcafee00) [inline 332-ci-manager/config/ci-manager-config.yml:369]: this change is temp, there's an issue with building the proper tags in harness.drone.io. Don't merge until we fix this.

**Comment by eoinmcafee00** [comment]
trigger uts

**Comment by eoinmcafee00** [comment]
trigger uts

**Comment by Raghav-Murali** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/pod/CIOidcCredentialDetails.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+package io.harness.delegate.beans.ci.pod;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class CIOidcCredentialDetails {
```
Comment (harshjain12) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/pod/CIOidcCredentialDetails.java:15]: Use interface

File: 879-pipeline-ci-commons/src/main/java/io/harness/utils/GcpOidcAuthenticator.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    return gcpOidcTokenRequestDTO;
+  }
+
+  private String fetchOidcIdToken(GcpOidcTokenRequestDTO gcpOidcTokenRequestDTO) throws IOException {
+    Response<ResponseDTO<String>> oidcIdToken =
+        connectorResourceClient.getOidcIdToken(gcpOidcTokenRequestDTO).execute();
+    if (oidcIdToken.isSuccessful()) {
+      return oidcIdToken.body().getData();
+    } else {
+      throw new IOException("Failed to fetch OIDC ID Token: " + oidcIdToken.code() + " " + oidcIdToken.message());
```
Comment (harshjain12) [inline 879-pipeline-ci-commons/src/main/java/io/harness/utils/GcpOidcAuthenticator.java:63]: We should not log secret

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/pod/CIOidcCredentialDetails.java (hunk 2 above)
Comment (eoinmcafee00) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/pod/CIOidcCredentialDetails.java]: class no longer needed, deleted.

File: 879-pipeline-ci-commons/src/main/java/io/harness/utils/GcpOidcAuthenticator.java (hunk 3 above)
Comment (eoinmcafee00) [reply 879-pipeline-ci-commons/src/main/java/io/harness/utils/GcpOidcAuthenticator.java]: removed.

**Comment by eoinmcafee00** [comment]
trigger uts

**Comment by Raghav-Murali** [comment]
trigger uts

**Review by jamie-harness — APPROVED** [review]

**Comment by Raghav-Murali** [comment]
trigger uts

**Comment by Raghav-Murali** [comment]
trigger uts

**Review by eoinmcafee00 — DISMISSED** [review]

**Comment by tiwaritejas** [comment]
trigger uts

**Comment by eoinmcafee00** [comment]
trigger uts

**Comment by eoinmcafee00** [comment]
trigger uts

**Comment by eoinmcafee00** [comment]
trigger uts

**Comment by gn-harness** [comment]
FMing as unrelated tests are failing

**Pull request merged**